content-src/components/DiscoveryStreamComponents/DSImage/DSImage.jsx

```diff
@@ -0,0 +1,107 @@
+import {cache} from "./cache";
+import React from "react";
+import ReactDOM from "react-dom";
+
+export class DSImage extends React.PureComponent {
+  constructor(props) {
+    super(props);
+
+    this.onOptimizedImageError = this.onOptimizedImageError.bind(this);
+
+    this.state = {
+      isSeen: false,
+      optimizedImageFailed: false,
+    };
+  }
+
+  onSeen(element) {
+    if (this.state && element[0].isIntersecting) {
+      if (this.props.optimize) {
+        this.setState({
+          containerWidth: ReactDOM.findDOMNode(this).clientWidth,
+        });
+      }
+
+      this.setState({
+        isSeen: true,
+      });
+
+      // Stop observing since element has been seen
+      this.observer.unobserve(ReactDOM.findDOMNode(this));
+    }
+  }
+
+  reformatImageURL(url, width) {
+    // Change the image URL to request a size tailored for the parent container width
+    // Also: force JPEG, quality 60, no upscaling, no EXIF data
+    // Uses Thumbor: https://thumbor.readthedocs.io/en/latest/usage.html
+    return `https://img-getpocket.cdn.mozilla.net/${width}x0/filters:format(jpeg):quality(60):no_upscale():strip_exif()/${encodeURIComponent(url)}`;
+  }
+
+  componentDidMount() {
+    let options = {
+      root: document.querySelector(`document`),
+      threshold: 0, // Load as soon as the first pixel crosses into view
```

Comment:
Heh, I think 0 is also the default here too, so we can remove this one. I'm sure about this one, not sure about root.